src/components/ContentUnit/tabs/associations/CUModal.js

```diff
@@ -24,12 +24,23 @@ class CUModal extends ListWithCheckboxBase {
     this.askForData(1);
   }
 
-  componentWillReceiveProps(nextProps) {
-    const { wipAssociate } = this.props;
-    if (nextProps.unit && wipAssociate && !nextProps.wipAssociate) {
-      this.askForData(1);
-      this.handleClose();
+  static getDerivedStateFromProps(props, state) {
+    if (!props.unit) {
+      return null;
+    }
+
+    const { wipAssociate, fetchList, handleToggleModal } = props;
+    if (wipAssociate) {
+      return { wip: true };
     }
+
+    if (state.wip && !wipAssociate) {
+      fetchList(NS_UNIT_ASSOCIATION_CU, 1);
```

Comment:
This comments repeats everywhere you changed `componentWillReceiveProps` 

`getDerivedStateFromProps` is not the place for network calls and other side effects.
You are advised to use `componentDidUpdate`

Please read:
https://reactjs.org/docs/react-component.html#static-getderivedstatefromprops
https://reactjs.org/docs/react-component.html#componentdidupdate

And consider each case on its own

Reply:
https://github.com/Bnei-Baruch/mdb-admin/pull/62/commits/07965e1c25758e3192cf86921586d2317037ade1